### PR TITLE
Make auto-import pages support .tsx files.

### PR DESCRIPTION
**Pull request by peterp**
`require.resolve` did not support `.tsx` extensions. So, now we're using glob instead. Fast, not recursive. Nice!

TODO:
- [x] Use fixtures directory for tests.
- [x] Remove recursion.
- [x] Make sure `consts` names are PascalCase.

@dac09 Probably helpful for your work on pre-rendering.

### CODE REVIEW

**Comment by Tobbe** [comment]
> `require.resolve` did not support `.tsx` extensions.

Because we're using node's resolver, and not TS's, right?

Anyway, this is great stuff! Now we don't have to wrap our tsx pages in a .js wrapper just to auto-load them with the router!

**Comment by peterp** [comment]
@Tobbe 
> Because we're using node's resolver, and not TS's, right?

I think this has something to do with `require.extensions` (which is deprecated), for awhile `.ts` and `.tsx` were available, but I'm not sure when they got removed.

I figured that babel's TS module would have registered those in nodejs.

https://nodejs.org/api/modules.html#modules_require_extensions

**Comment by dac09** [comment]
@peterp this looks great. 


Just to check in case there's a simpler way, did you try adding extensions to the Babel preset file? I noticed in my pre-render script that typescript files weren't being loaded until I added the line 

`extensions: ['.tsx', '.ts', '.jsx', '.js'], `

As part of the Babel config.

**Comment by peterp** [comment]
@dac09 Crazy. I would be _very_ surprised if that's actually something that works. It's something that we've always wanted... because otherwise it has to be specified via the CLI, which is super annoying.

I can't seem to find it in the docs and I found a bunch of issues [and a PR](https://github.com/babel/babel/pull/12216), but it doesn't seem like it merged.

If that does work, awesome, but I think this is a better solution, since I removed the recursion and opted for glob instead.

File: packages/internal/src/__tests__/paths.test.ts [hunk 1]
```diff
@@ -5,14 +5,26 @@ import { processPagesDir, resolveFile, ensurePosixPath } from '../paths'
 describe('paths', () => {
   describe('processPagesDir', () => {
     it('it accurately finds the pages', () => {
-      const pagesDir = path.resolve(__dirname, './fixtures/web/src/pages')
+      const pagesDir = path.resolve(
+        __dirname,
+        '../../../../__fixtures__/example-todo-main/web/src/pages'
+      )
```
Comment (peterp) [inline packages/internal/src/__tests__/paths.test.ts:11]: I should test the component names as well.

**Pull request merged**